### PR TITLE
H-344: Implement "Settings" pages layout

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

This PR implements the sidebar and layout for settings pages.

As part of this it also:
- Adds the root 'Organizations' settings page (a table of the user's orgs)
- Restores the 'create new organization' and 'settings' links to the workspace switcher (fixes H-186)
- Fixes breadcrumbs so that (a) items that don't go to valid pages aren't links, and (b) links are client-side rather than server requests (fixes H-290)
- Makes the main body background for all pages except user-made pages gray, and fixes the centering of them
- Renames the route from `/account` to `/settings`
- Renames 'workspace' to 'organization' where we are talking about organizations specifically

## 🔗 Related links

- [Figma design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH?type=design&node-id=8022-265030&mode=dev) (internal)

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

- There are a couple of styling issues with the sidebar 
    – at the moment org names which are too wide are ellided, rather than wrapping. A bit more work is needed to make sure the 'expand' caret appears right next to wrapped text, and since @nonparibus is doing more thinking about how expanding is working, I will wait to see the outcome of that before investing more time in it.
    - similarly, the designs call for the highlight indicator bar at the left of the sidebar to be cut off short at the top and bottom. I started doing this by adjusting margin and padding on the fly, but this leads to layout jank when things animate in/out. The bar probably needs to be built in a more complex way that can wait until we know what will happen with expansion buttons

## 🐾 Next steps

For orgs, other outstanding work:
- Allow setting an org logo
- When creating an org, it doesn't show up in the user's list of orgs without a refresh
- The 'Create organization' form should live in the settings layout and there should be a button from the 'Organizations' page to get to it

## 🛡 What tests cover this?

- None

## ❓ How to test this?

1.  Visit settings and click around

## 📹 Demo

https://github.com/hashintel/hash/assets/37743469/94622f6b-ae48-48f9-aa3c-bc975f099122

### CODE REVIEW

File: apps/hash-api/src/integrations/linear/webhook.ts [hunk 1]
```diff
@@ -54,8 +54,8 @@ export const linearWebhook: RequestHandler<{}, string, string> = async (
   }
 
   if (
-    tupleIncludes(["create", "update"] as const, payload.action) &&
-    tupleIncludes(["Issue", "User"] as const, payload.type)
+    tupleIncludes(["create", "update"], payload.action) &&
+    tupleIncludes(["Issue", "User"], payload.type)
```
Comment (CiaranMn) [inline apps/hash-api/src/integrations/linear/webhook.ts:58]: driveby: I've updated the function so that callers of it don't need to do this

File: apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page.tsx [hunk 2]
```diff
@@ -235,12 +235,10 @@ const Page: NextPageWithLayout = () => {
                   crumbs={[
                     {
                       title: "Types",
-                      href: "#",
                       id: "types",
                     },
                     {
                       title: `${entityTypeIsLink ? "Link" : "Entity"} Types`,
-                      href: "#",
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page.tsx:243]: These links can be restored in the 'view all types' PR

File: apps/hash-frontend/src/pages/change-password.page.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import {
 } from "./shared/ory-kratos";
 import { useKratosErrorHandler } from "./shared/use-kratos-flow-error-handler";
 
-const SettingsPage: NextPageWithLayout = () => {
+const ChangePasswordPage: NextPageWithLayout = () => {
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/change-password.page.tsx:17]: To free up the route... is this actually used anywhere?

File: apps/hash-frontend/src/pages/settings/organizations/[shortname]/members.page.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  return (
+    <>
+      <NextSeo title={`${org.name} | Members`} />
+
+      <OrgSettingsContainer
+        header={org.name}
+        sectionLabel="Members"
+        ref={topRef}
+      >
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/settings/organizations/[shortname]/members.page.tsx:60]: This and the layout function at the bottom are the only changes in this and other existing pages that appear deleted but have been moved

File: apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-page-wrapper/entity-page-header.tsx [hunk 5]
```diff
@@ -52,7 +51,7 @@ export const EntityPageHeader = ({
 
       {editBar}
 
-      <Box py={3.75}>
+      <Box py={3.75} sx={{ background: "white" }}>
```
[the remark this answers is not in the document]
Comment (nonparibus) [reply apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-page-wrapper/entity-page-header.tsx:54]: I think this is good practice anyway. Things like our "black" are not actually #000 in the design system, for example.

File: apps/hash-frontend/src/pages/settings/organizations/index.page.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  const router = useRouter();
+
+  const topRef = useRef<HTMLSpanElement>(null);
+
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  if (!authenticatedUser.accountSignupComplete) {
+    void router.push("/");
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/settings/organizations/index.page.tsx:23]: Really I'd like to gate these routes on the server - loading in the user on the server would also cut down on loading jank for bits relying on the user / org. If we stick with client-side gating something central would be good, yes

File: apps/hash-frontend/src/pages/shared/top-context-bar.tsx [hunk 7]
```diff
@@ -58,6 +58,7 @@ export const TopContextBar = ({
     <Box
       sx={[
         {
+          background: "white",
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/shared/top-context-bar.tsx]: Addressed in https://github.com/hashintel/hash/pull/2884/commits/2e2df719c9a4754f74751ee5be54e4fd6039ed07

File: apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-page-wrapper/entity-page-header.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-page-wrapper/entity-page-header.tsx]: Addressed in https://github.com/hashintel/hash/pull/2884/commits/2e2df719c9a4754f74751ee5be54e4fd6039ed07

File: apps/hash-frontend/src/pages/settings/shared/settings-layout.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+const generateMenuLinks = (organizations: Org[]): SidebarItemData[] => {
+  const organizationItems: SidebarItemData[] = organizations
+    .sort((a, b) => a.name.localeCompare(b.name))
+    .map((org) => [
+      {
+        label: org.name,
+        href: `/settings/organizations/${org.shortname}`,
+        parentHref: "/settings/organizations",
+      },
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/settings/shared/settings-layout.tsx]: See https://github.com/hashintel/hash/pull/2884#discussion_r1290465139

File: libs/@hashintel/block-design-system/src/icons/caret-down.tsx [hunk 9]
```diff
@@ -10,7 +13,10 @@ export const CaretDownIcon: FunctionComponent<SvgIconProps> = (props) => {
       fill="none"
       {...props}
     >
-      <path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z" />
+      <path
+        fill={fill}
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply libs/@hashintel/block-design-system/src/icons/caret-down.tsx]: Thank you, I've fixed this in https://github.com/hashintel/hash/pull/2884/commits/9ad916604bfb4847e144d36a7b8dc94f7f297adc

It wasn't working because I was passing `fill` directly to `CaretDownIcon` instead of inside `sx`. Moving it into `sx` fixed it

File: apps/hash-frontend/src/shared/icons/people-group-icon.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <SvgIcon
+      width="640"
+      height="512"
+      viewBox="0 0 640 512"
+      fill="none"
+      {...props}
+    >
+      <path
+        fill={fill}
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/shared/icons/people-group-icon.tsx]: Replied there

**Review by benwerner01 — CHANGES_REQUESTED** [review]
Thanks for addressing my comments, this PR looks good to me. I have spotted one remaining issue which I think got introduced with the most recent fixes, where the sidebar collapses when visiting the "Members" page:

https://github.com/hashintel/hash/assets/42802102/b72f930a-0ef0-4b4b-87f7-95e324768ea6

**Comment by CiaranMn** [comment]
> Thanks for addressing my comments, this PR looks good to me. I have spotted one remaining issue which I think got introduced with the most recent fixes, where the sidebar collapses when visiting the "Members" page:

thanks @benwerner01, fixed in https://github.com/hashintel/hash/pull/2884/commits/c3f2fc8047e720b621724bd7c312127bbc041be8

**Review by benwerner01 — DISMISSED** [review]
Thanks Ciaran, looks good to me 👍

**Review by benwerner01 — APPROVED** [review]
